samza-core/src/main/java/org/apache/samza/clustermanager/ContainerAllocator.java

```diff
@@ -203,23 +213,31 @@ void assignResourceRequests() {
           state.matchedResourceRequests.incrementAndGet();
         }
 
-        containerManager.handleContainerLaunch(request, preferredHost, peekAllocatedResource(preferredHost), resourceRequestState, this);
+        boolean containerLaunchComplete =
+            containerManager.handleContainerLaunch(request, preferredHost, peekAllocatedResource(preferredHost),
+                resourceRequestState, this);
+
+        /**
+         * Some Container launch requests are due to Container Placement actions like move, restarts. Under those
+         * circumstances a container launch needs to wait for the previous container incarnation to stop. In this scenario
+         * the allocator thread needs to sleep and recheck the stop again
+         */
+        if (!containerLaunchComplete) {
+          break;
```

Comment:
Do you want to log this outcome?

Reply:
This will cause log spam when allocator thread wakes up to check if the active container has stopped.